### PR TITLE
Update test_baseexception.py from cpython 3.11.2

### PR DESCRIPTION
**Pull request by dalinaum**
#4564

### CODE REVIEW

**Comment by fanninpm** [comment]
Please break any RustPython-related lines into a separate commit. This makes the `git blame` easier to keep track of.

**Comment by dalinaum** [comment]
@fanninpm Sure, I just split the previous commit into two commits.

**Review by youknowone — APPROVED** [review]
Thank you!

File: Lib/test/test_baseexception.py [hunk 1]
```diff
@@ -28,8 +30,9 @@ def test_inheritance(self):
             except TypeError:
                 pass
 
-        inheritance_tree = open(os.path.join(os.path.split(__file__)[0],
-                                                'exception_hierarchy.txt'))
+        inheritance_tree = open(
+                os.path.join(os.path.split(__file__)[0], 'exception_hierarchy.txt'),
+                encoding="utf-8")
```
[the remark this answers is not in the document]
Comment (dalinaum) [reply Lib/test/test_baseexception.py:35]: I uploaded it as a PR #4627.

**Pull request merged**